pkg/services/ngalert/notifier/channels/testing.go

```diff
@@ -2,12 +2,61 @@ package channels
 
 import (
 	"context"
+	"encoding/base64"
+	"os"
+	"testing"
 	"time"
 
 	"github.com/grafana/grafana/pkg/models"
 	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
 )
 
+// deleteFunc deletes the fake image.
+type deleteFunc func()
+
+// newFakeImageStore creates a fake image on disk and returns an image
+// with a token, path and URL. The test should call deleteFunc to delete
+// the image from disk at the end of the test.
+func newFakeImageStore(t *testing.T) (ImageStore, deleteFunc) {
```

Comment:
Same question as in the other review - if you are never reading `b` from disk, can you use the other Fake included in the package already?

Presumably what this gives you is a non-failing `os.Stat` call in contexts where that's useful.

Reply:
Like this? https://github.com/grafana/grafana/pull/51013